.github/workflows/advanced.yml

```diff
@@ -105,29 +105,39 @@ jobs:
       - name: Get workflow SHA
         id: workflow-sha
         run: |
-          echo sha=$(git log -1 --pretty=format:"%H" .github/workflows/ .github/scripts) >> "$GITHUB_OUTPUT"
+          .github/scripts/get-files-sha.sh ${{ github.workspace }} .github/workflows/advanced.yml
 
       - name: Get binutils SHA
         id: binutils-sha
         run: |
-          .github/scripts/get-sha.sh ${{ env.BINUTILS_REPO }} ${{ env.BINUTILS_BRANCH }}
+          .github/scripts/get-repository-sha.sh ${{ env.BINUTILS_REPO }} ${{ env.BINUTILS_BRANCH }}
```

Comment:
I would expect to merge repo sha with binutils scripts sha, the same for gcc and mingw

Reply:
I see, it was done differently in the workflow. Ok

Reply:
The source code and the scripts for building it are two separate things. If the hash of those two hashes would be used as a key, first, it would obfuscate what particular revision of the binutils sources and what the particular revision of the scripts were used and, second, it would make caching less effective.

Reply:
I meant to combine them together, not to make another hash on top

Reply:
> I meant to combine them together, not to make another hash on top

Then I don't understand what you mean exactly. Can you be a little bit more specific, please?

Reply:
I expected that "get binutils sha" step will be getting sha repo, sha scripts and combine them together. Then I saw binutils scripts sha is calculated separately and included in cache name. Which is fine for me.